deploy/clickhouse-keeper/clickhouse-keeper-1-node-256M-for-test-only.yaml

```diff
@@ -0,0 +1,246 @@
+  name: clickhouse-keeper
+  labels:
+    app: clickhouse-keeper
+spec:
+  selector:
+    matchLabels:
+      app: clickhouse-keeper
+  serviceName: clickhouse-keepers
+  replicas: 1
+  updateStrategy:
+    type: RollingUpdate
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: clickhouse-keeper
+        what: node
+      annotations:
+        prometheus.io/port: '7000'
+        prometheus.io/scrape: 'true'
+    spec:
+# affinity removed to allow use in single node test environment
+#      affinity:
+#        podAntiAffinity:
+#          requiredDuringSchedulingIgnoredDuringExecution:
+#            - labelSelector:
+#                matchExpressions:
+#                  - key: "app"
+#                    operator: In
+#                    values:
+#                      - clickhouse-keeper
+#              topologyKey: "kubernetes.io/hostname"
+      volumes:
+        - name: clickhouse-keeper-settings
+          configMap:
+            name: clickhouse-keeper-settings
+            items:
+              - key: keeper_config.xml
+                path: keeper_config.xml
+      containers:
+        - name: clickhouse-keeper
+          imagePullPolicy: IfNotPresent
+          image: "clickhouse/clickhouse-server:latest"
+          resources:
+            requests:
+              memory: "256M"
+              cpu: "1"
+            limits:
+              memory: "4Gi"
+              cpu: "2"
+          volumeMounts:
+            - name: clickhouse-keeper-settings
+              mountPath: /etc/clickhouse-keeper/
+            - name: clickhouse-keeper-datadir-volume
+              mountPath: /var/lib/clickhouse-keeper
+          env:
+            - name: SERVERS
+              value: "1"
+            - name: RAFT_PORT
+              value: "9234"
+          command:
+            - bash
+            - -x
+            - -c
+            - |
+              HOST=`hostname -s` &&
+              DOMAIN=`hostname -d` &&
+              if [[ $HOST =~ (.*)-([0-9]+)$ ]]; then
+                  NAME=${BASH_REMATCH[1]}
+                  ORD=${BASH_REMATCH[2]}
+              else
+                  echo "Failed to parse name and ordinal of Pod"
+                  exit 1
+              fi &&
+              export MY_ID=$((ORD+1)) &&
+              mkdir -p /tmp/clickhouse-keeper/config.d/ &&
+              {
+                echo "<yandex><keeper_server>"
+                echo "<server_id>${MY_ID}</server_id>"
+                echo "<raft_configuration>"
+                for (( i=1; i<=$SERVERS; i++ )); do
+                    echo "<server><id>${i}</id><hostname>$NAME-$((i-1)).${DOMAIN}</hostname><port>${RAFT_PORT}</port></server>"
+                done
+                echo "</raft_configuration>"
+                echo "</keeper_server></yandex>"
+              } >> /tmp/clickhouse-keeper/config.d/generated-keeper-settings.xml &&
+              cat /tmp/clickhouse-keeper/config.d/generated-keeper-settings.xml && 
```

Comment:
btw - why /tmp?

Reply:
Cause double mount ConfigMap + temporary generated config in the same folder `/etc/clickhouse-keeper` + `/etc/clickhouse-keeper/config.d/` not possible in Kubernetes

Reply:
I think we can mount single file /etc/clickhouse-keeper/keeper_config.xml as a volume. 

Storing configs in /tmp is not a good practice (due to security reasons).

Reply:
I tried, even when I mount only one key from ConfigMap with read/write mode, I can't dynamically write into config.d/ subfolder

/tmp/ is ok in this case, cause we create this config on every restart, before starting `keeper` daemon itself